### PR TITLE
Add default domain to API response

### PR DESCRIPTION
**Pull request by Salamandar**
## The problem

The web admin interface needs to know the default domain to select it in the drop down lists. (app install).

https://github.com/YunoHost/issues/issues/1633

...

## Solution

Add a `default` value that contains the index of the default domain in the list.

...

## PR Status
Needs another PR on yunohost-admin repo.
...

## How to test

Just try to install a new app, the domain selection list should be alphabetically sorted and the main domain should be selected.
...

### CODE REVIEW

File: src/yunohost/domain.py [hunk 1]
```diff
@@ -61,7 +61,11 @@ def domain_list(exclude_subdomains=False):
                 continue
         result_list.append(domain)
 
-    return {'domains': result_list}
+    result_list.sort()
```
Comment (Psycojoker) [inline src/yunohost/domain.py:64]: You can use `sorted(result_list)` bellow to save a line if you want (but that won't be a in-place sort)

Comment (Salamandar) [reply src/yunohost/domain.py:64]: I think I prefer as it is, because result_list is used twice below (and we don't want sort() to run twice, do we ? ;) )

**Comment by zamentur** [comment]
I am not sure the cli behaviour is what we want:
```
root@ynh:/ynh-dev# yunohost domain list
default: 0
domains: ynh.local
```
In cli we might put default: ynh.local or just nothing... What's you point of view ?

File: src/yunohost/domain.py [hunk 2]
```diff
@@ -61,7 +61,11 @@ def domain_list(exclude_subdomains=False):
...
+    return {
+        'domains': result_list,
+        'default': result_list.index(_get_maindomain())
```
Comment (zamentur) [inline src/yunohost/domain.py:67]: I think this index research should be done in web admin to improve cli display AND avoid postinstall failure (it search the index of 'yunohost.org' which is not in domains list).
```suggestion
        'default': _get_maindomain()
```

**Review by zamentur — CHANGES_REQUESTED** [review]

**Comment by zamentur** [comment]
Linked to https://github.com/YunoHost/yunohost-admin/pull/306

**Comment by zamentur** [comment]
We should merge this way to sort domain i think: https://github.com/YunoHost/yunohost/pull/860/files

File: src/yunohost/domain.py (hunk 1 above)
Comment (alexAubin) [reply src/yunohost/domain.py:64]:
```suggestion
```

Domain sorting is to be addressed by https://github.com/YunoHost/yunohost/pull/860

File: src/yunohost/domain.py [hunk 3]
```diff
@@ -61,7 +61,10 @@ def domain_list(exclude_subdomains=False):
                 continue
         result_list.append(domain)
 
-    return {'domains': result_list}
+    return {
+        'domains': result_list,
+        'default': _get_maindomain()
```
Comment (alexAubin) [inline src/yunohost/domain.py:66]:
```suggestion
        'main': _get_maindomain()
```

The semantic "main" should be kept (though iirc we kind of use "default domain" in places in the webadmin views but in cli we have for example `yunohost tools main-domain` etc... )

**Pull request merged**